front/src/modules/Items/ItemList.tsx

```diff
@@ -1,57 +1,117 @@
   onOpenItem: (item: WarehouseEntryWithPath) => void;
   onStartCutting: (item: WarehouseEntryWithPath) => void;
-  cutting: null | {item: WarehouseEntryWithPath};
+  cutting: null | { item: WarehouseEntryWithPath };
   isSearch: boolean;
 }) {
   const [newItemName, setNewItemName] = useState("");
+  const [slicedArray, setSlicedArray] = useState<WarehouseEntryWithPath[]>([]);
+  const [refState, setRefState] = useState(false);
+  const [isObserving, setIsObserving] = useState(false);
+  const observerRef = useRef<IntersectionObserver | null>(null);
+
+  const listLength = props.list.length;
+  const currentIndexRef = useRef(20);
+  const oldIndexRef = useRef(0);
+
+  useEffect(() => {
+    if (slicedArray.length > listLength) {
+      setSlicedArray(props.list.slice(0, currentIndexRef.current));
+      currentIndexRef.current = Math.min(20, listLength);
+      oldIndexRef.current = 0;
+      setRefState(false);
+    }
+    if (refState) {
+      if (slicedArray.length < listLength) {
+        const currentIndex = currentIndexRef.current;
+        const oldIndex = oldIndexRef.current;
+
+        setSlicedArray(prev => [
+          ...prev,
+          ...props.list.slice(oldIndex, currentIndex),
+        ]);
+
+        oldIndexRef.current = currentIndex;
+        currentIndexRef.current = Math.min(currentIndex + 10, listLength);
+
+        setRefState(false);
+      }
+    }
+  }, [listLength, refState, props.list, slicedArray]);
 
   function handleCreateItem() {
-    props.onCreateItem({name: newItemName});
+    props.onCreateItem({ name: newItemName });
     setNewItemName("");
   }
 
-  const Row = ({index, style}: {index: number; style: React.CSSProperties}) => {
-    const item = props.list[index];
-    return (
-      <div style={style}>
-        <Item
-          key={item.id}
-          isSearch={props.isSearch}
-          item={item}
-          onDelete={() => props.onDeleteItem(item)}
-          onGoForward={() => props.onOpenItem(item)}
-          onEdit={props.onUpdateItem}
-          onCutStart={() => props.onStartCutting(item)}
-          cutting={props.cutting}
-        />
-      </div>
-    );
-  };
+  useEffect(() => {
+    const options = {
+      root: document.querySelector("#root"),
+      rootMargin: "0px",
+      threshold: 0.7,
+    };
+
+    if (!observerRef.current) {
+      observerRef.current = new IntersectionObserver(entries => {
+        const [entry] = entries;
+        if (entry.isIntersecting && !isObserving) {
+          setIsObserving(true);
+          setRefState(true);
+        }
+      }, options);
+    }
+
+    const target = document.querySelector(".endOfList");
+    if (target && observerRef.current) {
+      observerRef.current.observe(target);
+    }
+
+    return () => {
+      if (observerRef.current) {
+        observerRef.current.disconnect();
+      }
+    };
+  }, [isObserving]);
+
+  useEffect(() => {
+    if (refState === false) {
+      setIsObserving(false);
+    }
+  }, [refState]);
```

Comment:
I don't understand, this seems like refState = isObserving. Why do you have two variables for the same data? And for some reason you set it in a useEffect.
I'm also not sure what either of those variable do. You might need to name them better.